### DISCUSSION TITLE
Error response from daemon: unauthorized when I'm trying to pull public image from GPR

### DISCUSSION THREAD

**Discussion by webmakaka** [community/community#22032] [New to GitHub]
<p>Hello experts!</p>
<p>I’m study Github Actions + Github Package Registry.</p>
<p>And i created images in Github Package Registry.</p>
<p>Here are my packages.</p>
<p><a href="https://github.com/webmakaka?tab=packages" rel="nofollow noopener">https://github.com/webmakaka?tab=packages</a></p>
<p>On page</p>
<p><a href="https://github.com/webmakaka/MERN-Stack-Front-To-Back-v2.0/packages/22520" rel="nofollow noopener">https://github.com/webmakaka/MERN-Stack-Front-To-Back-v2.0/packages/22520</a></p>
<p>I see the command to pull image.</p>
<pre><code class="lang-auto">
docker pull docker.pkg.github.com/webmakaka/mern-stack-front-to-back-v2.0/api.anketa.info:latest

</code></pre>
<p>When i am trying to run pull command on my linux host.</p>
<pre><code class="lang-auto">
$ docker pull docker.pkg.github.com/webmakaka/mern-stack-front-to-back-v2.0/api.anketa.info:latest  
Error response from daemon: unauthorized: Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.

</code></pre>
<p>What should i do to get public image without any authorization?</p>
<p>Thanks!</p>
<p>P.S. I was looking for solution for 15 minutes before create a new topic.</p>

**Comment by webmakaka** [comment]
<p><em>Happy Halloween Day!</em></p>

**Comment by clarkbw** [comment]
<p>Right now, even public images, require authorization.  Include your PAT as your password and your GitHub username as your username for Docker</p>

**Comment by yogig** [comment]
<p>Thank you, SOLVED!!</p>

**Comment by rogithub** [comment]
<p>Could you provide an example on how to do that?</p>

**Comment by ARM-Source** [comment]
<p>Can you provide an example? </p>

**Comment by barkardk** [comment]
<p>Create a an access token that you can use from your github account <br>
github -&gt; settings-&gt;tokens -&gt; Personal access tokens </p>
<p>Make sure the token has Read and/or Write access to github registry </p>
<p>Then login to the registry </p>
<p>docker login <a href="http://docker.pkg.github.com" rel="nofollow noopener">docker.pkg.github.com</a> --username &lt;your_user_name&gt; --password &lt;generated_token_not_password&gt;</p>
<p>and now pull/push to the registry should work </p>

**Comment by fatihyildizhan** [comment]
<p>I have the same problem. I created an access token with full permission but, it didn’t work too.</p>
<p>Do you have any suggestion? Thank you.</p>
<p>I am only able to push. When I try to pull it I am getting this error:</p>
<p><code>Error response from daemon: unauthorized: Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.</code></p>

**Comment by webmakaka** [comment, marked as answer]
<p>Everything is changed since i asked this question.<br>
Everything works.</p>

**Comment by eve000017** [comment]
<p>I success to print out “Login Succeeded” in Bash when I login with the code:</p>
<p>docker login <a href="http://docker.pkg.github.com" rel="noopener nofollow ugc">docker.pkg.github.com</a> -u USERNAME -p TOKEN</p>
<p>But, when I pull the image, the error pop out.</p>
<p>“Error response from daemon: unauthorized: Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.”</p>
<p>my code:<br>
sudo docker pull <a href="http://docker.pkg.github.com/USERNAME/github-actions-for-packages/tic-tac-toe:f29" rel="noopener nofollow ugc">docker.pkg.github.com/USERNAME/github-actions-for-packages/tic-tac-toe:f29</a></p>
<p>How to solve this issue?</p>
<p>Thanks<br>
Regards,<br>
Eve</p>

**Comment by airtower-luna** [comment]
<p>The <a href="https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-docker-registry">old Docker registry at <code>docker.pkg.github.com</code></a> has been replaced by <a href="https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry"><code>ghcr.io</code></a>. According to <a href="https://docs.github.com/en/packages/working-with-a-github-packages-registry/migrating-to-the-container-registry-from-the-docker-registry">Migrate to Container registry</a> the image should have been migrated automatically. Try using <code>ghcr.io</code> instead.</p>
